akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -621,7 +622,12 @@ private[remote] class AssociationRegistry(createAssociation: Address ⇒ Associa
   private[this] val associationsByAddress = new AtomicReference[Map[Address, Association]](Map.empty)
   private[this] val associationsByUid = new AtomicReference[ImmutableLongMap[Association]](ImmutableLongMap.empty)
 
-  @tailrec final def association(remoteAddress: Address): Association = {
+  final def association(remoteAddress: Address): Option[Association] = {
+    val currentMap = associationsByAddress.get
+    currentMap.get(remoteAddress)
+  }
```

Comment:
An Association can be created because you send to it. How can this work?
`AssociationRegistry` will throw when a new Association is created, via the `newAssociation.associate()` call, which calls the run method. So I don't think there is a need for check on every send